js/data/data_source/data_source.js

```diff
@@ -806,6 +835,26 @@ var DataSource = Class.inherit({
         });
     },
 
+    _onPush: function(changes) {
+        if(this._reshapeOnPush) {
+            this.load();
```

Comment:
There is no changes argument in the changed event in this case.

Reply:
Also it will be good to have changes argument in the customizeStoreLoadOptions event to invalidate cache for the DataGrid widget.